FireboltDotNetSdk.Tests/Unit/TypesConverterTest.cs

```diff
@@ -140,6 +151,60 @@ public void ConvertInvalidValue()
         Assert.That(exception!.Message, Is.EqualTo("Input string was not in a correct format."));
     }
 
+    public static IEnumerable<TestCaseData> ArrayTestCases()
+    {
+        yield return new TestCaseData(new[] { 1, 2 }, new[] { 1, 2 }, "array(int)");
+        yield return new TestCaseData(new[] { "1", "2" }, new[] { 1, 2 }, "array(int)");
+        yield return new TestCaseData(new[] { "2022-05-10", "2022-05-11" }, new[] { DateTime.Parse("2022-05-10"), DateTime.Parse("2022-05-11") }, "array(date)");
+        yield return new TestCaseData(new[] { "2022-05-10 23:01:02", "2022-05-11 23:01:02" }, new[] { DateTime.Parse("2022-05-10 23:01:02"), DateTime.Parse("2022-05-11 23:01:02") }, "array(timestamp)");
+        yield return new TestCaseData(new[] { new[] { 1, 2 }, new[] { 3, 4 } }, new[] { new[] { 1, 2 }, new[] { 3, 4 } }, "array(array(int))");
+    }
+
+    [TestCaseSource(nameof(ArrayTestCases))]
+    public void ConvertArray(object arrayValue, object expected, String type)
+    {
+        ColumnType columnType = ColumnType.Of(type);
+        object? result = TypesConverter.ConvertToCSharpVal(arrayValue, columnType);
+        Assert.That(result, Is.EqualTo(expected));
+    }
+
+    public static IEnumerable<TestCaseData> StructTestCase()
+    {
+        yield return new TestCaseData(
+            new Dictionary<String, object> { { "a", 1 }, { "b", "value" } },
+            new Dictionary<String, object> { { "a", 1 }, { "b", "value" } },
+            "struct(a int, b string)"
+        );
```

Comment:
Let's add some nulls and columns with spaces, just to check that converting value works in those cases.